MekHQ/docs/DistributedMekHQ.md

```diff
@@ -0,0 +1,310 @@
+- If a Remote Campaign cannot support the Host Campaign version the connection is terminated.
+- If the Remote Campaign does not use the correct password the connection is terminated.
+
+### Security
+As Distributed MekHQ uses gRPC, TLS will be made available as an option to secure the underlying transport. Further, a ConnectionRequest may include an optional password. Messages sent over the streams will use Protobuf and not support arbitrary serialization.
+
+### ConnectionRequest
+The ConnectionRequest message is sent by the Remote Campaign when it first connects to the Host Campaign.
+
+The ConnectionRequest message has the following properties:
+- Remote Campaign Version
+- Remote Campaign Name
+- Remote Campaign UUID
+- Remote Campaign Date
+- Password (*OPTIONAL*)
+
+The Host Campaign shall ignore additional ConnectRequest messages once a connection has been established for a Remote Campaign.
+
+### ConnectionResponse
+The ConnectionResponse message is sent by the Host Campaign when it accepts a connection from a Remote Campaign.
+
+The ConnectionResponse message has the following properties:
+- Host Campaign Version
+- Host Campaign Name
+- Host Campaign UUID
+- Host Campaign Date
+- Remote Campaign List
+    - Remote Campaign Name
+    - Remote Campaign UUID
+    - Remote Campaign Date
+    - Remote Campaign Team
+
+### Message Bus Messages
+The Message Bus contains all client-to-server messages to be processed by the server and either redistributed or transformed.
+
+Each message is encapsulated over a bidirectional gRPC stream with the following format:
+- Timestamp
+- Message Type
+- Metadata Map (String -> String)
+- Message Bytes (protobuf)
+
+#### CampaignDateChanged
+The CampaignDateChanged message notifies the Host Campaign that a Remote Campaign has advanced days.
+
+The CampaignDateChanged message has the following properties:
+- Campaign UUID
+- Campaign Date
+
+This message is redistributed to all connected campaigns.
+
+#### GMModeChanged
+The GMModeChanged message notifies the Host Campaign that a Remote Campaign has changed their GM Mode.
+
+The GMModeChanged message has the following properties:
+- Campaign UUID
+- Is GM?
+
+This message is redistributed to all connected campaigns.
+
+#### TeamAssignment
+The TeamAssignment message notifies a Remote Campaign of team assignments.
+
+The TeamAssignment message has the following properties:
+- Team List
+    - Remote Campaign UUID
+    - Remote Campaign Team
+
+The Host Campaign is always Team 1.
+
+This message is only sent by the Host Campaign.
+
+#### DailyLog
+The DailyLog message notifies the Host Campaign of a daily log activity.
+
+The DailyLog message has the following properties:
+- Campaign UUID
+- Log Date
+- Log List
+    - Log Entry
+
+This message may be sent at any point in a day and may be sent mulitple times, but implementations could delay sending this message until a day has completed.
```

Comment:
multiple times :wink:

Reply:
For speeling and grammer feel free to use suggested edits.

Reply:
I was on my phone and didn't know how to on that device